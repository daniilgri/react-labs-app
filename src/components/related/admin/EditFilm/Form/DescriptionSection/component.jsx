import React from "react";
import PropTypes from "prop-types";
import Tags from "./Tags";

import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
  TagsField,
  Textarea,
} from "./styles";

const Component = ({
  onChange,
  title,
  description,
  tags,
  errors,
  onTagAdd,
  onTagDelete,
}) => {
  return (
    <Wrapper>
      <Head>
        <Title>Add description info</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="title">Title</Label>
          <Input
            id="title"
            name="title"
            placeholder="title"
            type="text"
            onChange={onChange}
            value={title}
          />
          {errors.title && <ErrorText>{errors.title}</ErrorText>}
        </Field>
        <Field>
          <Label htmlFor="description">Description</Label>
          <Textarea
            id="description"
            name="description"
            placeholder="..."
            type="text"
            onChange={onChange}
            value={description}
          />
          {errors.description && <ErrorText>{errors.description}</ErrorText>}
        </Field>
        <TagsField>
          <Tags
            onSet={onTagAdd}
            onDelete={onTagDelete}
            values={tags}
            errors={errors}
          />
        </TagsField>
      </Body>
    </Wrapper>
  );
};

Component.defaultProps = {
  onChange: () => {},
  title: "",
  description: "",
  tags: [],
  errors: {},
  onTagAdd: () => {},
  onTagDelete: () => {},
};

Component.propTypes = {
  onChange: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  tags: PropTypes.array.isRequired,
  errors: PropTypes.object.isRequired,
  onTagAdd: PropTypes.func.isRequired,
  onTagDelete: PropTypes.func.isRequired,
};

export default Component;
