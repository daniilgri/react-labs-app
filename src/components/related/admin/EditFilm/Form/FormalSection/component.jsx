import React from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
} from "./styles";

const Component = ({ onChange, values, errors }) => {
  return (
    <Wrapper>
      <Head>
        <Title>Formal</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="ticketPrice">Ticket price</Label>
          <Input
            id="ticketPrice"
            name="ticketPrice"
            placeholder="price"
            type="text"
            onChange={onChange}
            value={values.ticketPrice}
          />
          {errors.ticketPrice && <ErrorText>{errors.ticketPrice}</ErrorText>}
        </Field>
      </Body>
    </Wrapper>
  );
};

Component.defaultProps = {
  onChange: () => {},
  values: {},
  errors: {},
};

Component.propTypes = {
  onChange: PropTypes.func.isRequired,
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
};

export default withRouter(Component);
