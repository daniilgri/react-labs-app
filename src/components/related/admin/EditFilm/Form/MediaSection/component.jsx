import React from "react";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";

import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
  ImagePreview,
} from "./styles";

const Component = ({ onImageUpload, errors, previewImage }) => {
  return (
    <Wrapper>
      <Head>
        <Title>Add media info</Title>
      </Head>
      <Body>
        <ImagePreview alt="preview" src={previewImage} />
        <Field>
          <Label htmlFor="imageAsFile">Image</Label>
          <Input
            id="imageAsFile"
            name="imageAsFile"
            type="file"
            onChange={onImageUpload}
          />
          {errors.imageAsFile && <ErrorText>{errors.imageAsFile}</ErrorText>}
        </Field>
      </Body>
    </Wrapper>
  );
};

Component.defaultProps = {
  onImageUpload: () => {},
  errors: {},
  previewImage: "",
};

Component.propTypes = {
  onImageUpload: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired,
  previewImage: PropTypes.string.isRequired,
};
export default withRouter(Component);
