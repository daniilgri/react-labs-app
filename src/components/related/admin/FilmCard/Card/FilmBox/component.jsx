import React from "react";
import PropTypes from "prop-types";
import { v4 as uuidv4 } from "uuid";

import { Wrapper, ImageWrapper, InfoList, Title, Description, Price, Tags, OneTag } from "./styles";

const component = ({ title, description, image, ticketPrice, tags }) => (
  <Wrapper>
    <ImageWrapper src={image} />
    <InfoList>
      <Tags>
        {tags.map(el => (
          <OneTag key={uuidv4()}>{el}</OneTag>
        ))}
      </Tags>
      <Price>{ticketPrice} $</Price>
      <Title>{title}</Title>
      <Description>{description}</Description>
    </InfoList>
  </Wrapper>
);

component.defaultProps = {
  title: "",
  description: "",
  image: "",
  ticketPrice: "0",
  tags: [],
};

component.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  ticketPrice: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.string),
};

export default component;
