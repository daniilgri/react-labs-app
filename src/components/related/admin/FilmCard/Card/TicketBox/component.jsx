import React from "react";
import PropTypes from "prop-types";
import { v4 as uuidv4 } from "uuid";
import { withRouter } from "react-router-dom";

import {
  Wrapper,
  Title,
  Content,
  OptionList,
  OptionTitle,
  ButtonOption,
  Box,
  TimesList,
  TimeOptionButton,
  DateValue,
} from "./styles";

const Component = ({ screeningDates }) => {
  return (
    <Wrapper>
      <Title>Ticketing</Title>
      <Content>
        <Box>
          <OptionTitle>Day</OptionTitle>
          <OptionList>
            {screeningDates.length > 0 &&
              screeningDates.map(dateItem => (
                <ButtonOption key={uuidv4()}>
                  <DateValue>{dateItem.date}</DateValue>
                  <TimesList>
                    {dateItem.times.map(timeItem => (
                      <TimeOptionButton key={uuidv4()}>{timeItem}</TimeOptionButton>
                    ))}
                  </TimesList>
                </ButtonOption>
              ))}
          </OptionList>
        </Box>
      </Content>
    </Wrapper>
  );
};

Component.propTypes = {
  screeningDates: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default withRouter(Component);
