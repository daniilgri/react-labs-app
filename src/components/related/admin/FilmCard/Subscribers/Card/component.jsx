import React from "react";
import PropTypes from "prop-types";

import {
  Wrapper,
  Image,
  InfoSection,
  NameSection,
  EmailAddress,
  Controllers,
  FilledButton,
  Button,
} from "./styles";

const component = ({ user, onConfirmDeleteRequest, onDelete }) => {
  const handleConfirmDeleteRequestButtonOnClick = (event) => {
    event.preventDefault();

    onConfirmDeleteRequest({ uid: user.uid });
  };

  const handleDeleteButtonOnClick = (event) => {
    event.preventDefault();

    onDelete({ uid: user.uid });
  };

  return (
    <Wrapper>
      <Image src="https://picsum.photos/800/1200" />
      <InfoSection>
        <NameSection>
          {user.firstName} {user.lastName}
        </NameSection>
        <EmailAddress>{user.email}</EmailAddress>
        <Controllers>
          <FilledButton
            disabled={!user.requestOnDelete}
            onClick={handleConfirmDeleteRequestButtonOnClick}
          >
            Confirm delete request
          </FilledButton>
          <Button onClick={handleDeleteButtonOnClick} color="#ff6868">
            Delete user
          </Button>
        </Controllers>
      </InfoSection>
    </Wrapper>
  );
};

component.defaultProps = {
  user: {},
  onConfirmDeleteRequest: () => {},
  onDelete: () => {},
};

component.propTypes = {
  user: PropTypes.object.isRequired,
  onConfirmDeleteRequest: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default component;
