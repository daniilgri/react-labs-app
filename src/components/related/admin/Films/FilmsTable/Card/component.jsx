import React from "react";
import PropTypes from "prop-types";

import {
  Wrapper,
  Image,
  InfoSection,
  Title,
  ScreeningDate,
  Controllers,
  Button,
} from "./styles";
const component = ({ film }) => (
  <Wrapper>
    <Image src="https://picsum.photos/800/1200" />
    <InfoSection>
      <Title>{film.title}</Title>
      <Controllers>
        <Button color="#000">Edit</Button>
        <Button color="#ff6868">Delete</Button>
      </Controllers>
    </InfoSection>
  </Wrapper>
);

export default component;
