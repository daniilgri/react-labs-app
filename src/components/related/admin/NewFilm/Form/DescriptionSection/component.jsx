import React from "react";
import PropTypes from "prop-types";
import Tags from "./Tags";

import {
  Wrapper,
  Head,
  Title,
  Body,
  Field,
  Label,
  Input,
  ErrorText,
  TagsField,
  Textarea,
} from "./styles";

const Component = ({ onChange, values, errors, onTagAdd, onTagDelete }) => {
  return (
    <Wrapper>
      <Head>
        <Title>Add description info</Title>
      </Head>
      <Body>
        <Field>
          <Label htmlFor="title">Title</Label>
          <Input
            id="title"
            name="title"
            placeholder="title"
            type="text"
            onChange={onChange}
            value={values.title}
          />
          {errors.title && <ErrorText>{errors.title}</ErrorText>}
        </Field>
        <Field>
          <Label htmlFor="description">Description</Label>
          <Textarea
            id="description"
            name="description"
            placeholder="..."
            type="text"
            onChange={onChange}
            value={values.description}
          />
          {errors.description && <ErrorText>{errors.description}</ErrorText>}
        </Field>
        <TagsField>
          <Tags onSet={onTagAdd} onDelete={onTagDelete} values={values.tags} errors={errors} />
        </TagsField>
      </Body>
    </Wrapper>
  );
};

Component.propTypes = {
  onChange: PropTypes.func.isRequired,
  values: PropTypes.objectOf(
    PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.array])
  ).isRequired,
  errors: PropTypes.objectOf(PropTypes.object).isRequired,
  onTagAdd: PropTypes.func.isRequired,
  onTagDelete: PropTypes.func.isRequired,
};

export default Component;
