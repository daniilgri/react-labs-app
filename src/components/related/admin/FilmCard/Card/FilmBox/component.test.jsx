import React from "react";
import renderer from "react-test-renderer";

import component from "./component";

describe("admin: FilmCard Card FilmBox component", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <component
          film={{
            title: "",
            description: "",
            tags: [],
            image: "",
            rates: [],
            screeningDates: [],
            ticketPrice: "",
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
