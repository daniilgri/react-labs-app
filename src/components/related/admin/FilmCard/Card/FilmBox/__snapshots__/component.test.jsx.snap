// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin: FilmCard Card FilmBox component renders correctly 1`] = `
<component
  film={
    Object {
      "description": "",
      "image": "",
      "rates": Array [],
      "screeningDates": Array [],
      "tags": Array [],
      "ticketPrice": "",
      "title": "",
    }
  }
/>
`;
